tests/foreman/ui/test_contentcredentials.py

```diff
@@ -90,6 +90,61 @@ def test_positive_end_to_end(session, module_org, gpg_content):
         assert session.contentcredential.search(new_name)[0]['Name'] != new_name
 
 
+@pytest.mark.tier2
+def test_positive_block_delete_key_in_use(session, module_org, gpg_content):
+    """Create a new gpg key with valid content. Create a new product, assigning
+       the gpg key, and create an associated repository. Attempt to delete the
+       gpg key in use.
+
+    :id: b79fd3ff-8cdb-4cdd-94e5-76de742ec967
+
+    :expectedresults: deletion of the gpg key is blocked by raising TypeError,
+        the key is not deleted, and remains associated with the product and repository
+
+    :BZ: 2052904
+
+    :CaseLevel: Integration
+    """
+
+    name = gen_string('alpha')
+
+    with session:
+        # Create new gpg key with valid content
+        session.contentcredential.create(
+            {
+                'name': name,
+                'content_type': CONTENT_CREDENTIALS_TYPES['gpg'],
+                'content': gpg_content,
+            }
+        )
+        assert session.contentcredential.search(name)[0]['Name'] == name
+        gpg_key = entities.ContentCredential(organization=module_org).search(
+            query={'search': f'name="{name}"'}
+        )[0]
+        # Create new product with the gpg key, and a single associated new repository
+        product = entities.Product(gpg_key=gpg_key, organization=module_org).create()
+        repo = entities.Repository(product=product, url=settings.repos.yum_1.url).create()
+        # Assert the gpg key is associated with the new product and repo
+        values = session.contentcredential.read(gpg_key.name)
+        assert values['details']['name'] == name
+        assert values['details']['content_type'] == CONTENT_CREDENTIALS_TYPES['gpg']
+        assert values['products']['table'][0]['Used as'] == CONTENT_CREDENTIALS_TYPES['gpg']
+        assert values['repositories']['table'][0]['Used as'] == CONTENT_CREDENTIALS_TYPES['gpg']
```

Comment:
I would move all of this above `with session:` and use the API to create all of this stuff since it's test setup. Essentially the only thing we have to test with the UI here is trying to delete. Even better you could look and see if there are fixtures that already do all of this for you.

Reply:
Great thanks for the suggestion there, I'll work on getting this changed

Reply:
Just pushed changes with api coverage